src/bin/pgcopydb/pgcmd.c

```diff
@@ -350,8 +350,14 @@ pg_dump_db(PostgresPaths *pgPaths,
 
 	SafeURI safeURI = { 0 };
 	bool pgpassword_found_in_env = env_exists("PGPASSWORD");
-	char PGPASSWORD[MAXCONNINFO] = { 0 };
-
+	size_t len = strlen(pguri) + 1;
+	char *PGPASSWORD = (char *) calloc(len, sizeof(char));
+	safeURI.password = (char *) calloc(len, sizeof(char));
+	if (PGPASSWORD  == NULL || safeURI.password  == NULL)
+	{
+		log_error(ALLOCATION_FAILED_ERROR);
+		return false;
+	}
```

Comment:
When do you use the memory allocated here? I think you're using `strdup` which does its own memory allocation anyway, right?